cmd/ketch/builder_list_test.go

```diff
@@ -0,0 +1,72 @@
+
+	"github.com/shipa-corp/ketch/cmd/ketch/configuration"
+)
+
+const (
+	defaultBuilders = `VENDOR                 IMAGE                             DESCRIPTION
+Google:                gcr.io/buildpacks/builder:v1      GCP Builder for all runtimes                                                              
+Heroku:                heroku/buildpacks:18              heroku-18 base image with buildpacks for Ruby, Java, Node.js, Python, Golang, & PHP       
+Heroku:                heroku/buildpacks:20              heroku-20 base image with buildpacks for Ruby, Java, Node.js, Python, Golang, & PHP       
+Paketo Buildpacks:     paketobuildpacks/builder:base     Small base image with buildpacks for Java, Node.js, Golang, & .NET Core                   
+Paketo Buildpacks:     paketobuildpacks/builder:full     Larger base image with buildpacks for Java, Node.js, Golang, .NET Core, & PHP             
+Paketo Buildpacks:     paketobuildpacks/builder:tiny     Tiny base image (bionic build image, distroless run image) with buildpacks for Golang     
+`
+	userBuilders = `VENDOR                 IMAGE                             DESCRIPTION
+Google:                gcr.io/buildpacks/builder:v1      GCP Builder for all runtimes                                                              
+Heroku:                heroku/buildpacks:18              heroku-18 base image with buildpacks for Ruby, Java, Node.js, Python, Golang, & PHP       
+Heroku:                heroku/buildpacks:20              heroku-20 base image with buildpacks for Ruby, Java, Node.js, Python, Golang, & PHP       
+Paketo Buildpacks:     paketobuildpacks/builder:base     Small base image with buildpacks for Java, Node.js, Golang, & .NET Core                   
+Paketo Buildpacks:     paketobuildpacks/builder:full     Larger base image with buildpacks for Java, Node.js, Golang, .NET Core, & PHP             
+Paketo Buildpacks:     paketobuildpacks/builder:tiny     Tiny base image (bionic build image, distroless run image) with buildpacks for Golang     
+test vendor:           test image                        test description                                                                          
+`
+)
+
+func TestBuilderList(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		cfg      config
+		expected string
+	}{
+		{
+			name: "default values",
+			cfg: &mocks.Configuration{
+				KetchConfigObj: configuration.KetchConfig{},
+			},
+			expected: defaultBuilders,
+		},
+		{
+			name: "include user's builders",
+			cfg: &mocks.Configuration{
+				KetchConfigObj: configuration.KetchConfig{
+					AdditionalBuilders: []configuration.AdditionalBuilder{
+						{
+							Vendor:      "test vendor",
+							Image:       "test image",
+							Description: "test description",
+						},
+					},
+				},
+			},
+			expected: userBuilders,
+		},
+	}
+
+	for _, tt := range tests {
+		var buff bytes.Buffer
+		cmd := newBuilderListCmd(tt.cfg, &buff)
+		cmd.SetArgs([]string{})
```

Comment:
Maybe this is not required.

Reply:
I'm not sure what you mean. Do you mean that you don't think any tests are needed or that there is no need for two tests since testing to see if any additional builders are added also tests the defaults being printed?

Reply:
Ah no no. I am asking if `cmd.SetArgs([]string{})` is required if we are setting a zero value there? Will it show the same behaviour if we remove this line?

Reply:
Sorry, I misunderstood. Removing the `SetArgs` will result in the tests failing. According to the [documentation](https://pkg.go.dev/github.com/spf13/cobra?utm_source=godoc#Command.SetArgs) if `SetArgs` is not specified `os.Args[1:]` will be used. When I tried removing the line and running the tests, they failed with various unknown flags that got picked up from somewhere.